### PR TITLE
Various CNN Fixes and SqueezeNet Test

### PR DESCRIPTION
**Pull request by tjingrant**

### CODE REVIEW

**Comment by tjingrant** [comment]
I did a couple of fixes:

- conv and pooling; specifically paddings and data formats. Would you please help me fix the tests?
- I'm surprised to find out conv has a third param bias which is currently not documented.
- fixed a couple of default params.

Now squeeze net checks off.

File: test/test_model_large.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+import unittest
+import numpy as np
+import tensorflow as tf
+import onnx
+from onnx_tf.backend import run_model, prepare
+from onnx import helper
+from onnx.onnx_pb2 import TensorProto
+
+class TestNode(unittest.TestCase):
+  MODEL_PATH = "../../onnx_models/"
```
Comment (arpith-jacob) [inline test/test_model_large.py:15]: Instead of assuming a path, can we download the models?

File: onnx_tf/backend.py [hunk 2]
```diff
@@ -208,6 +206,10 @@ class TensorflowBackend(Backend):
   }
 
   @classmethod
+  def pad2tfpad(cls, pads):
+    return "VALID" if pads[0] == 0 else "SAME"
```
Comment (arpith-jacob) [inline onnx_tf/backend.py]: I didn't understand the ONNX pads parameter from the documentation.  Why does it have multiple dimensions?

File: onnx_tf/backend.py [hunk 3]
```diff
@@ -442,7 +443,8 @@ def handle_batch_normalization(cls, node, input_dict):
   @classmethod
   def handle_concat(cls, node, input_dict):
     values = [input_dict[a] for a in node.inputs]
-    axis = node.attrs["axis"]
+    # apparently this is what's needed for squeezenet to work
+    axis = node.attrs["axis"] if "axis" in node.attrs.keys() else 1
```
Comment (arpith-jacob) [inline onnx_tf/backend.py]: axis = node.attrs.get("axis", 1)

File: onnx_tf/backend.py [hunk 4]
```diff
@@ -464,11 +466,12 @@ def _conv(cls, node, input_dict, transpose=False):
...
     if transpose:
-      # Translate weights from (M x C x KH x KW) to (KH x KW X C X M)
-      perm = list(range(2, weights_rank)) + [1, 0]
-    else:
       # Translate weights from (C x M x KH x KW) to (KH x KW X C X M)
       perm = list(range(2, weights_rank)) + [0, 1]
+    else:
+      # Translate weights from (M x C x KH x KW) to (KH x KW X C X M)
+      perm = list(range(2, weights_rank)) + [1, 0]
+
```
Comment (arpith-jacob) [inline onnx_tf/backend.py:475]: Looks good.  You may also have to change the tests for conv and convtranspose.

File: onnx_tf/backend.py [hunk 5]
```diff
@@ -484,9 +487,11 @@ def _conv(cls, node, input_dict, transpose=False):
       warnings.warn("Unsupported kernel_shape attribute by Tensorflow in "
```
Comment (arpith-jacob) [inline onnx_tf/backend.py:484]: You can remove the 'pads' warning since the pads attribute is being used.

File: onnx_tf/backend.py [hunk 6]
```diff
@@ -484,9 +487,11 @@ def _conv(cls, node, input_dict, transpose=False):
...
                     "Conv operator. The attribute will be ignored.",
                     UserWarning)
-    return [tf.nn.convolution(x, weights, "SAME", strides=strides,
+    tf_pads = cls.pad2tfpad(node.attrs["pads"])
+    output = tf.nn.bias_add(tf.nn.convolution(x, weights, tf_pads, strides=strides,
                               dilation_rate=dilations,
-                              data_format=data_format)]
+                              data_format=data_format), input_dict[node.inputs[2]], data_format="NCHW")
```
Comment (arpith-jacob) [inline onnx_tf/backend.py]: Can you document the third parameter?  Also, have bias = input_dict[node.inputs[2]] at the top and use 'bias' in the call to tf.nn.bias_add().

File: onnx_tf/backend.py (hunk 2 above)
Comment (tjingrant) [reply onnx_tf/backend.py]: it has 2 times the spatial dims number of elements. 2 for each spacial axis indicating pads in the beginning/end of this dimension.

Comment (arpith-jacob) [reply onnx_tf/backend.py]: Does tensorflow support adding such fine-grained control for padding?  It doesn't seem like it, from the documentation.

File: onnx_tf/backend.py (hunk 6 above)
Comment (tjingrant) [reply onnx_tf/backend.py]: This documentation is coming in the latest PR.

File: onnx_tf/backend.py (hunk 2 above)
Comment (tjingrant) [reply onnx_tf/backend.py]: No it does not, this pad2tfpad follows a simple logic: if the first element is 0 then return "VALID" which is usually no padding and if not return "SAME" which has padding. So far it works. We will eventually remove it in favor of the auto_pad attribute in the latest PR.

File: test/test_model_large.py (hunk 1 above)
Comment (tjingrant) [reply test/test_model_large.py:15]: yeah i think shell script is better here since we have to unpack a compressed file.

**Comment by arpith-jacob** [comment]
LGTM
